src/dbus_api/api/manager_3_4/methods.rs

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use dbus::{Message, Path};
+use dbus_tree::{MTSync, MethodInfo, MethodResult};
+use futures::executor::block_on;
+
+use crate::{
+    dbus_api::{
+        blockdev::create_dbus_blockdev,
+        filesystem::create_dbus_filesystem,
+        pool::create_dbus_pool,
+        types::{DbusErrorEnum, TData, OK_STRING},
+        util::{engine_to_dbus_err_tuple, get_next_arg, tuple_to_option},
+    },
+    engine::{Engine, Name, Pool, PoolIdentifier, PoolUuid, StartAction, UnlockMethod},
+    stratis::StratisError,
+};
+
+pub fn start_pool<E>(m: &MethodInfo<'_, MTSync<TData<E>>, TData<E>>) -> MethodResult
+where
+    E: 'static + Engine,
+{
+    let base_path = m.path.get_name();
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+    let dbus_context = m.tree.get_data();
+    let default_return: (
+        bool,
+        (Path<'static>, Vec<Path<'static>>, Vec<Path<'static>>),
+    ) = (false, (Path::default(), Vec::new(), Vec::new()));
+    let return_message = message.method_return();
+
+    let id_str: &str = get_next_arg(&mut iter, 0)?;
+    let unlock_method = {
```

Comment:
Need to reverse order in which values are unmarshaled...id_type before unlock_method.